oak/proto/manager.proto

```diff
@@ -20,126 +20,51 @@ package oak;
 
 // An ApplicationConfiguration represents a unit of deployment in Oak.
 //
-// It defines a connectivity graph using unidirectional Channels between Nodes.
-//
-// The following constraints apply to the Application configuration:
-// - at most one GrpcServerNode
-// - at most one LogNode
+// An Oak Application is built from a collection of interconnected Nodes,
+// each of which is running the code described by an entry in this
+// configuration.  These Nodes are created dynamically at runtime, with
+// the exception of the specified initial Node (which is created by the
+// Oak runtime).
 message ApplicationConfiguration {
-  repeated Node nodes = 1;
-  repeated Channel channels = 2;
-  repeated WasmContents wasm_contents = 3;
+  // Collection of available Node contents, indexed by name (which must
+  // be unique across the collection).  Each Node in the application will
+  // run code that is identified by an entry in this collection.
+  repeated NodeContents contents = 1;
+  // Indication of what contents the initial Node should run.  Must identify a
+  // NodeContents entry that holds a WebAssemblyCode object.
+  string initial_node = 2;
 }
 
-// A Node represents a unit of computation in Oak.
-message Node {
-  // The name of the Node. Must be unique within the Application.
-  string node_name = 1;
-
-  oneof node_type {
-    LogNode log_node = 2;
-    GrpcServerNode grpc_server_node = 3;
-    WebAssemblyNode web_assembly_node = 4;
-    StorageProxyNode storage_node = 5;
+// NodeContents indicate what code a created Node should run.
+message NodeContents {
+  string name = 1;
+  oneof code_type {
+    WebAssemblyCode wasm_code = 2;
+    LogCode log_code = 3;
```

Comment:
Why the `_code` suffix for these nodes, and the `code_` prefix for the oneof? I was thinking of these things as node configurations rather than "code". In fact my personal suggestion would be something like:
- `ApplicationConfiguration` contains `NodeConfiguration`s, and that contains `WebAssemblyConfiguration`, `LogConfiguration`, `StorageProxyConfiguration` etc.

WDYT?

Reply:
Yeah, that's a better wording that my current version &ndash; I'll shift things across.

Reply:
Let me know when I should look at it again (please re-request review so it appears on my list of PRs waiting for review)

Reply:
Renaming done.